eland/ml/pytorch/external_models.py

```diff
@@ -0,0 +1,391 @@
+
+    def __init__(self, model_id: str, task_type: str):
+        self._model_id = model_id
+        self._task_type = task_type.replace('-', '_')
+
+        # Elasticsearch model IDs need to be a specific format: no special chars, all lowercase, max 64 chars
+        self._es_model_id = self._model_id.replace('/', '__').lower()[:64]
+
+        # load Hugging Face model and tokenizer
+        self._tokenizer = transformers.AutoTokenizer.from_pretrained(self._model_id)
+
+        # check for a supported tokenizer
+        if not isinstance(self._tokenizer, SUPPORTED_TOKENIZERS):
+            raise TypeError(f"Tokenizer type {self._tokenizer} not supported, must be one of: {SUPPORTED_TOKENIZERS_STR}")
+
+        self._traceable_model = self._create_traceable_model()
+        self._traced_model = self._traceable_model.trace()
+        self._vocab = self._load_vocab()
+        self._config = self._create_config()
+
+    @staticmethod
+    def _dict_to_ordered_list(dict: Dict[Any, Any], sort_by_key: bool = False, sort_by_value: bool = False) -> List[Any]:
+        assert not (sort_by_key and sort_by_value)
+        assert sort_by_key or sort_by_value
+
+        if sort_by_key:
+            sort_index = 0
+            select_index = 1
+        else:
+            sort_index = 1
+            select_index = 0
+
+        l = list(dict.items())
+        l.sort(key=lambda x: x[sort_index])
+        new_list = [x[select_index] for x in l]
+        return new_list
+
+    def _load_vocab(self):
+        vocabulary = HuggingFaceTransformerModel._dict_to_ordered_list(self._tokenizer.get_vocab(), sort_by_value=True)
+        return {
+            'vocabulary': vocabulary,
+        }
+
+    def _create_config(self):
+        do_lower_case = hasattr(self._tokenizer, 'do_lower_case') and self._tokenizer.do_lower_case
+
+        inference_config = {
+            self._task_type: {
+                'tokenization': {
+                    'bert': {
+                        'do_lower_case': do_lower_case,
+                    }
+                }
+            }
+        }
+
+        if hasattr(self._tokenizer, 'max_model_input_sizes'):
+            max_sequence_length = self._tokenizer.max_model_input_sizes.get(self._model_id)
+            if max_sequence_length:
+                inference_config[self._task_type]['tokenization']['bert']['max_sequence_length'] = max_sequence_length
+
+        if self._traceable_model.classification_labels():
+            inference_config[self._task_type]['classification_labels'] = self._traceable_model.classification_labels()
+
+        return {
+            'description': f"Model {self._model_id} for task type '{self._task_type}'",
+            'model_type': 'pytorch',
+            'inference_config': inference_config,
+            'input': {
+                'field_names': ['text_field'],
+            }
+        }
+
+    def _create_traceable_model(self) -> TraceableModel:
+
+        if self._task_type == 'fill_mask':
```

Comment:
nit: Some spacing between each of the `if-elif` blocks here would do a lot for readability.